lang/cs/Org.Apache.REEF.IMRU.Tests/TestTaskManager.cs

```diff
@@ -0,0 +1,679 @@
+            Assert.Throws<IMRUSystemException>(add);
+            add = () => taskManager.AddTask(MasterTaskId, MockConfig(), CreateMockActiveContext(1));
+            Assert.Throws<IMRUSystemException>(add);
+        }
+
+        /// <summary>
+        /// Tests invalid arguments when adding tasks
+        /// </summary>
+        [Fact]
+        public void NullArguments()
+        {
+            var taskManager = CreateTaskManager();
+            taskManager.AddTask(MasterTaskId, MockConfig(), CreateMockActiveContext(0));
+
+            Action add = () => taskManager.AddTask(MapperTaskIdPrefix + 1, null, CreateMockActiveContext(1));
+            Assert.Throws<IMRUSystemException>(add);
+
+            add = () => taskManager.AddTask(MapperTaskIdPrefix + 2, MockConfig(), null);
+            Assert.Throws<IMRUSystemException>(add);
+        }
+
+        /// <summary>
+        /// Tests passing invalid arguments in creating TaskManager
+        /// </summary>
+        [Fact]
+        public void InvalidArgumentsInCreatingTaskManger()
+        {
+            Action taskManager = () => CreateTaskManager(0, MasterTaskId);
+            Assert.Throws<IMRUSystemException>(taskManager);
+
+            taskManager = () => CreateTaskManager(1, null);
+            Assert.Throws<IMRUSystemException>(taskManager);
+        }
+
+        /// <summary>
+        /// Tests set tasks to complete for running tasks
+        /// </summary>
+        [Fact]
+        public void TestCompletingTasks()
+        {
+            var taskManager = TaskManagerWithTasksRunning();
+            Assert.True(taskManager.AreAllTasksInState(TaskState.TaskRunning));
+
+            taskManager.RecordCompletedTask(CreateMockCompletedTask(MapperTaskIdPrefix + 1));
+            taskManager.RecordCompletedTask(CreateMockCompletedTask(MapperTaskIdPrefix + 2));
+            taskManager.RecordCompletedTask(CreateMockCompletedTask(MasterTaskId));
+            Assert.True(taskManager.AreAllTasksInState(TaskState.TaskCompleted));
+        }
+
+        /// <summary>
+        /// Tests adding/closing running tasks
+        /// </summary>
+        [Fact]
+        public void TestAddingClosingRunningTasks()
+        {
+            var taskManager = TaskManagerWithTasksSubmitted();
+
+            taskManager.RecordRunningTask(CreateMockRunningTask(MasterTaskId));
+            taskManager.RecordRunningTask(CreateMockRunningTask(MapperTaskIdPrefix + 1));
+
+            var runningTask2 = CreateMockRunningTask(MapperTaskIdPrefix + 2);
+            taskManager.RecordRunningTaskDuringSystemFailure(runningTask2, TaskManager.CloseTaskByDriver);
+
+            taskManager.CloseAllRunningTasks(TaskManager.CloseTaskByDriver);
+            Assert.True(taskManager.AreAllTasksInState(TaskState.TaskWaitingForClose));
+        }
+
+        /// <summary>
+        /// Tests set tasks to fail for running tasks
```

Comment:
Replace by "Tests recording task failure for running tasks during Running Or SubmissionState"